pkg/signature/ecdsa.go

```diff
@@ -181,6 +181,11 @@ func (e ECDSAVerifier) VerifySignature(signature, message io.Reader, opts ...Ver
 		return fmt.Errorf("reading signature: %w", err)
 	}
 
+	// Without this check, VerifyASN1 panics on an invalid curve.
```

Comment:
is it worth a defensive check at the beginning of this function 

```
if e.publicKey == nil {
...
}
```

rare but you'd panic if that were the case

Reply:
Done